modules/kubernetes-addons/crossplane/main.tf

```diff
@@ -18,18 +18,20 @@ module "helm_addon" {
 # AWS Provider
 #--------------------------------------
 resource "kubectl_manifest" "aws_controller_config" {
-  count = var.aws_provider.enable == true ? 1 : 0
+  count = local.aws_provider.enable == true ? 1 : 0
   yaml_body = templatefile("${path.module}/aws-provider/aws-controller-config.yaml", {
-    iam-role-arn = "arn:${var.addon_context.aws_partition_id}:iam::${var.addon_context.aws_caller_identity_account_id}:role/${var.addon_context.eks_cluster_id}-${local.aws_provider_sa}-irsa"
+    iam-role-arn          = "arn:${var.addon_context.aws_partition_id}:iam::${var.addon_context.aws_caller_identity_account_id}:role/${var.addon_context.eks_cluster_id}-${local.aws_provider.name}-irsa"
```

Comment:
hmm - this looks to be prior to the change, but we should not re-construct the role arn and instead pass it. where is this role arn coming from?

Reply:
The role is for the aws-provider deployment.
I'm guessing the role is created by the irsa module below
https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/07bb3ca207ffb417a01ef6a8ffe6e019c5a27de0/modules/kubernetes-addons/crossplane/main.tf#L47
```hcl
module "aws_provider_irsa" {
  count                             = local.aws_provider.enable == true ? 1 : 0
  source                            = "../../../modules/irsa"
  create_kubernetes_namespace       = false
  create_kubernetes_service_account = false
  kubernetes_namespace              = local.namespace
  kubernetes_service_account        = "${local.aws_provider.name}-*"
  irsa_iam_policies                 = local.aws_provider.additional_irsa_policies
  irsa_iam_role_path                = var.addon_context.irsa_iam_role_path
  irsa_iam_permissions_boundary     = var.addon_context.irsa_iam_permissions_boundary
  eks_cluster_id                    = var.addon_context.eks_cluster_id
  eks_oidc_provider_arn             = var.addon_context.eks_oidc_provider_arn
  depends_on                        = [kubectl_manifest.aws_provider]
}
```

How do you proposed we handle the role? Is there a way to pass a role name to the irsa module, instead of constructing it?